app/models/user_invitation.rb

```diff
@@ -0,0 +1,20 @@
+class UserInvitation < ApplicationRecord
+  belongs_to :site
+  belongs_to :inviting_user, class_name: 'User'
+
+  validates :email, uniqueness: { scope: :site_id }
```

Comment:
Add reference to accepted_site_user and destroy invitations if site user is deleted. Nil inviting_user if inviting user is deleted.